src/test/scala/latis/service/hapi/DataServiceSpec.scala

```diff
@@ -167,4 +197,42 @@ class DataServiceSpec extends FlatSpec {
   it should "reject trailing commas" in {
     csvDecoderReject("a,b,")
   }
+
+  "The DataService" should "correctly generate a response for a CSV request" in {
+    val req = Request[IO](Method.GET, uri"/data?dataset=testdataset&start=2000-01-01&stop=2000-01-06&format=csv")
+    val resp = dataService.orNotFound(req)
+    (for {
+      body <- resp.flatMap { res =>
+        res.bodyText.compile.toList
+      }
+      contentType = resp.unsafeRunSync().headers.get(CIString("Content-Type")).get.head.value
+    } yield {
+      val testStr = "2000-01-01T00:00:00.000Z,1\n" +
+                    "2000-01-02T00:00:00.000Z,5\n" +
+                    "2000-01-03T00:00:00.000Z,4\n" +
+                    "2000-01-04T00:00:00.000Z,2\n" +
+                    "2000-01-05T00:00:00.000Z,0\n"
+      assert(body.mkString == testStr)
+      assert(contentType == "text/csv")
+    }).unsafeRunSync()
+  }
+
+  it should "correctly generate a response for a binary request" in {
+    val req = Request[IO](Method.GET, uri"/data?dataset=testdataset&start=2000-01-01&stop=2000-01-06&format=binary")
+    val resp = dataService.orNotFound(req)
+    (for {
+      body <- resp.flatMap { res =>
+        res.body.compile.toList
+      }
+      contentType = resp.unsafeRunSync().headers.get(CIString("Content-Type")).get.head.value
+    } yield {
+      val testBin = utf8enc(fs2.Stream("2000-01-01T00:00:00.000Z")).toList ::: intLenc(1).require.toByteArray.toList :::
+                    utf8enc(fs2.Stream("2000-01-02T00:00:00.000Z")).toList ::: intLenc(5).require.toByteArray.toList :::
+                    utf8enc(fs2.Stream("2000-01-03T00:00:00.000Z")).toList ::: intLenc(4).require.toByteArray.toList :::
+                    utf8enc(fs2.Stream("2000-01-04T00:00:00.000Z")).toList ::: intLenc(2).require.toByteArray.toList :::
+                    utf8enc(fs2.Stream("2000-01-05T00:00:00.000Z")).toList ::: intLenc(0).require.toByteArray.toList
```

Comment:
I think something like this would be cleaner:

```scala
val encoder = codecs.list(codecs.utf8 ~ codecs.int32L)
val testBin = encoder.encode(
  List(
    ("2000-01-01T00:00:00.000Z", 1),
    ("2000-01-02T00:00:00.000Z", 5),
    ("2000-01-03T00:00:00.000Z", 4),
    ("2000-01-04T00:00:00.000Z", 2),
    ("2000-01-05T00:00:00.000Z", 0)
  )
).require.bytes.toArray.toList
```

Reply:
I like that, definitely a lot easier to read, and it works the same